packages/dds/tree/src/test/tree/anchorSet.spec.ts

```diff
@@ -492,6 +492,49 @@ describe("AnchorSet", () => {
 		]);
 	});
 
+	it("childrenChangedAfterBatch event includes the changed fields", () => {
+		const fieldOne: FieldKey = brand("one");
+		const fieldTwo: FieldKey = brand("two");
+		const fieldThree: FieldKey = brand("three");
+
+		const anchors = new AnchorSet();
+
+		const anchor0 = anchors.track(makePath([rootFieldKey, 0]));
+		const node0 = anchors.locate(anchor0) ?? assert.fail();
+
+		const expectedChangedFields = new Set<FieldKey>([fieldOne, fieldTwo, fieldThree]);
+		let listenerFired = false;
+		node0.on("childrenChangedAfterBatch", ({ anchor, changedFields }) => {
+			// This is the main validation of this test
+			assert.deepEqual(changedFields, expectedChangedFields);
+			listenerFired = true;
```

Comment:
Nit: should we ensure that the listener is only called _once_?

Reply:
In fact, by applying the pattern Craig suggested to me before and is used in some of the other new tests. Keep a log where all event payloads are pushed when the event fires, and in the end check that both the correct number of entries are present and they have the expected structure.